Content.Server/Shuttles/Systems/ShuttleTimerSystem.cs

```diff
@@ -0,0 +1,98 @@
+        {
+            var timerXform = Transform(uid);
+
+            // no false positives.
+            if (timerXform.MapUid == null)
+                return;
+
+            string key;
+            args.Data.TryGetValue("ShuttleMap", out EntityUid? shuttleMap);
+            args.Data.TryGetValue("SourceMap", out EntityUid? source);
+            args.Data.TryGetValue("DestMap", out EntityUid? dest);
+            args.Data.TryGetValue("Docked", out bool docked);
+            string?[] text = new string?[] { docked ? "ETD" : "ETA" };
+
+            switch (timerXform.MapUid)
+            {
+                // sometimes the timer transforms on FTL shuttles have a hyperspace mapuid, so matching by grid works as a fallback.
+                case var local when local == shuttleMap || timerXform.GridUid == shuttleMap:
+                    key = "ShuttleTimer";
+                    break;
+                case var origin when origin == source:
+                    key = "SourceTimer";
+                    break;
+                case var remote when remote == dest:
+                    key = "DestTimer";
+                    text = new string?[] { "ETA" };
+                    break;
+                default:
+                    return;
+            }
+
+            if (!args.Data.TryGetValue(key, out TimeSpan duration))
+                return;
+
+            if (args.Data.TryGetValue("Text", out string?[]? label))
+                text = label;
+
+            _appearanceSystem.SetData(uid, TextScreenVisuals.TargetTime, _gameTiming.CurTime + duration);
+            _appearanceSystem.SetData(uid, TextScreenVisuals.ScreenText, text);
+        }
+
+        /// <summary>
+        /// Helper method for <see cref="RoundEndSystem"/> and <see cref="EmergencyShuttleSystem"/> to get the MapUids of
+        /// the `eshuttle`, `centcomm`, and the best-guess `station` (using <see cref="StationSystem.GetLargestGrid(StationDataComponent)"/>)
+        /// </summary>
+        /// <returns>{"eshuttle":uid?, "centcomm":uid?, "station":uid?}</returns>
+        public Dictionary<string, EntityUid?> GetEscapeMaps()
+        {
+            AllEntityQuery<StationEmergencyShuttleComponent>().MoveNext(out var station, out var eshuttleComp);
+            var targetGrid = _stationSystem.GetLargestGrid(Comp<StationDataComponent>(station));
+            var stationMap = targetGrid == null ? null : Transform(targetGrid.Value).MapUid;
+
+            var eshuttleMap = eshuttleComp?.EmergencyShuttle;
+
+            AllEntityQuery<StationCentcommComponent>().MoveNext(out var centcomm);
+            EntityUid? centcommMap = centcomm == null ? null : _mapManager.GetMapEntityId(centcomm.MapId);
+
+            return new Dictionary<string, EntityUid?>
+            {
+                {"eshuttle", eshuttleMap}, {"centcomm", centcommMap}, {"station", stationMap}
+            };
+        }
```

Comment:
This method feels weird but I can't articular how to make it better so it's fine as is.